api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -301,33 +324,32 @@ private boolean checkWorkspaceWriteAccess(String workspaceNamespace, String work
     }
     FirecloudWorkspaceId workspaceId = generateFirecloudWorkspaceId(workspace.getNamespace(),
         workspace.getName());
-    org.pmiops.workbench.db.model.Workspace existingWorkspace = workspaceService.get(
-        workspaceId.getWorkspaceNamespace(), workspaceId.getWorkspaceName());
+    org.pmiops.workbench.db.model.Workspace existingWorkspace = workspaceService.getByName(
+        workspaceId.getWorkspaceNamespace(), workspace.getName());
     if (existingWorkspace != null) {
       throw new BadRequestException(String.format(
           "Workspace %s/%s already exists",
           workspaceId.getWorkspaceNamespace(), workspaceId.getWorkspaceName()));
     }
-    try {
-      fireCloudService.createWorkspace(workspaceId.getWorkspaceNamespace(),
-          workspaceId.getWorkspaceName());
-    } catch (org.pmiops.workbench.firecloud.ApiException e) {
-      log.log(
-          Level.SEVERE,
-          String.format(
-              "Error creating FC workspace %s/%s: %s",
-              workspaceId.getWorkspaceNamespace(),
-              workspaceId.getWorkspaceName(),
-              e.getResponseBody()),
-          e);
-      // TODO: figure out what happens if the workspace already exists
-      throw new ServerErrorException("Error creating FC workspace", e);
+    FirecloudWorkspaceId attemptId = new FirecloudWorkspaceId(workspaceId.getWorkspaceNamespace(), workspaceId.getWorkspaceName());
+    for (int attemptValue = 1; attemptValue < 6; attemptValue++) {
+      attemptId =
```

Comment:
I would put this in the catch block. That way you don't assign attemptId above just to reassign it, and in the normal case there's no numeric suffix.